shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz_hints/presentation/StepQuizHintsFeature.kt

```diff
@@ -6,14 +6,15 @@ import org.hyperskill.app.reactions.domain.model.ReactionType
 import org.hyperskill.app.step.domain.model.CommentThread
 import org.hyperskill.app.step.domain.model.Step
 
-interface StepQuizHintsFeature {
-    companion object {
-        fun isHintsFeatureAvailable(step: Step): Boolean =
-            step.commentsStatistics.any { it.thread == CommentThread.HINT && it.totalCount > 0 }
-    }
-
-    sealed interface State {
-
+// Modifier 'internal' is not applicable inside 'interface' for StepQuizHintsFeature.State
+// Making StepQuizHintsFeature 'abstract class', 'sealed class' or 'object' solves this.
+// What's better to use?
```

Comment:
Usually objects are used for `context bounds` purposes. Only objects allowed to have static functions.

Reply:
OK, that makes sense. Let's stop on `object`.